jupyterlab_server/handlers.py

```diff
@@ -292,6 +304,34 @@ def add_handlers(web_app, config):
         handlers.append((
             workspace_api_path, WorkspacesHandler, workspaces_config))
 
+    # Handle local listings.
+    if config.listings_dir:
+
+        settings_config = web_app.settings.get('config', {}).get('LabServerApp', {})
+        blacklist_uris = settings_config.get('blacklist_uris', '')
+        whitelist_uris = settings_config.get('whitelist_uris', '')
+
+        if (blacklist_uris != '') and (whitelist_uris != ''):
+            raise Exception('Simultaneous blacklist_uris and whitelist_uris is not supported. Please define only one of those.')
+            # TODO(@echarles) Force exit here.
+            import sys
+            sys.exit(-1)
+
+        ListingsHandler.listings_refresh_ms = settings_config.get('listings_refresh_ms', 1000 * 60 * 5)
```

Comment:
See here, same thing about defaults. Above the default seems to be set to 1 minute and here you set to default of 5 minutes (which I believe will never be used because above default is set?)

Reply:
Updated the default. Same comment as above, the defaults are not processed as they could be.